### ISSUE TITLE
RuntimeError: cannot pin 'torch.cuda.HalfTensor' only dense CPU tensors can be pinned

### ISSUE THREAD

**Issue by birdup000** [lyogavin/Anima#81]
Running the Example code but with quantization 4bit on the latest version 2.5 

I get this issue

 logs:
 ```
 >>>> bitsandbytes installed
>>>> cache_utils installed
>>>> bitsandbytes installed
>>>> bitsandbytes installed
>>>> bitsandbytes installed
>>>> bitsandbytes installed
>>>> bitsandbytes installed
Fetching 25 files: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 25/25 [00:00<00:00, 49742.69it/s]
saved layers already found in /home/bird/.cache/huggingface/hub/models--garage-bAInd--Platypus2-70B-instruct/snapshots/b585e74bcaae02e52665d9ac6d23f4d0dbc81a0f/splitted_model.4bit
new version of transfomer, no need to use BetterTransformer, setting attn impl to sdpa...
attn imp: <class 'transformers.models.llama.modeling_llama.LlamaSdpaAttention'>
new version of transfomer, no need to use BetterTransformer, setting attn impl to sdpa...
attn imp: <class 'transformers.models.llama.modeling_llama.LlamaSdpaAttention'>
cuda:0:   0%|                                                                                                                                                          | 0/83 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/mnt/aedc235c-d458-41cb-88fa-1da70ef3b1bc/AirLLM/myAI.py", line 25, in <module>
    generation_output = model.generate(
  File "/home/bird/.local/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/bird/.local/lib/python3.10/site-packages/transformers/generation/utils.py", line 1718, in generate
    return self.greedy_search(
  File "/home/bird/.local/lib/python3.10/site-packages/transformers/generation/utils.py", line 2579, in greedy_search
    outputs = self(
  File "/home/bird/.local/lib/python3.10/site-packages/airllm/airllm.py", line 247, in __call__
    return self.forward(*args, **kwargs)
  File "/home/bird/.local/lib/python3.10/site-packages/airllm/airllm.py", line 313, in forward
    state_dict = future.result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/bird/.local/lib/python3.10/site-packages/airllm/airllm.py", line 187, in load_layer_to_cpu
    state_dict[k].pin_memory()
RuntimeError: cannot pin 'torch.cuda.HalfTensor' only dense CPU tensors can be pinned
```

**Comment by lyogavin** [comment]
I believe it's fixed in 21c01a3db2e4bbda19bb22b67b991b6adac15051

tested here:

https://github.com/lyogavin/Anima/blob/main/air_llm/tests/test_notebooks/test_compression.ipynb

**Comment by lyogavin** [comment]
sorry, the fix is in: 21c01a3db2e4bbda19bb22b67b991b6adac15051

airllm==2.6.1
